src/v/iceberg/snapshot_json.cc

```diff
@@ -0,0 +1,144 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/snapshot_json.h"
+
+#include "container/chunked_hash_map.h"
+#include "iceberg/json_utils.h"
+#include "iceberg/snapshot.h"
+#include "json/document.h"
+#include "model/timestamp.h"
+#include "strings/string_switch.h"
+
+namespace iceberg {
+
+namespace {
+
+constexpr std::string_view operation_to_str(snapshot_operation o) {
+    switch (o) {
+    case snapshot_operation::append:
+        return "append";
+    case snapshot_operation::replace:
+        return "replace";
+    case snapshot_operation::overwrite:
+        return "overwrite";
+    case snapshot_operation::delete_data:
+        return "delete";
+    }
+}
+
+snapshot_operation operation_from_str(const ss::sstring& operation_str) {
```

Comment:
nit: `using enum iceberg::snapshot_operation;` within this function scope might make these statements quite a bit shorter.

Reply:
Done